src/app/models/constraint-models/constraint.ts

```diff
@@ -18,12 +20,15 @@ export class Constraint {
   protected _parentConstraint: Constraint;
   // The negation flag indicating whether to add a logical negation to the constraints
   protected _negated: boolean;
+  // The dimension for which observations will be selected
+  private _dimension: string;
 
   constructor() {
     this.textRepresentation = '';
     this.mark = ConstraintMark.OBSERVATION;
```

Comment:
Looks to me that the `mark` can be removed. Its use is now superseded by the `dimension` attribute.

Reply:
we want to rename mark to dimension, not to add another dimension attribute, right?